Samples/PublicSamples/RecordingBot/src/RecordingBot.Services/ServiceSetup/AppHost.cs

```diff
@@ -73,30 +64,49 @@ public AppHost()
         /// <summary>
         /// Boots this instance.
         /// </summary>
-        public void Boot()
+        public void Boot(string[] args)
         {
+            var builder = WebApplication.CreateBuilder(args);
+
+            ServicePointManager.SecurityProtocol = SecurityProtocolType.Tls12;
+
             DotNetEnv.Env.Load(path: null, options: new DotNetEnv.LoadOptions());
+            builder.Configuration.AddEnvironmentVariables();
 
-            var builder = new ConfigurationBuilder();
+            // Load Azure Settings
+            var azureSettings = new AzureSettings();
+            builder.Configuration.GetSection(nameof(AzureSettings)).Bind(azureSettings);
+            azureSettings.Initialize();
+            builder.Services.AddSingleton(azureSettings);
 
-            // tell the builder to look for the appsettings.json file
-            builder
-                .AddJsonFile("appsettings.json", optional: true, reloadOnChange: true)
-                .AddEnvironmentVariables();
+            // Setup Listening Urls
+            builder.WebHost.UseKestrel(serverOptions =>
+            {
+                serverOptions.ListenAnyIP(azureSettings.CallSignalingPort + 1);
+                serverOptions.ListenAnyIP(azureSettings.CallSignalingPort, config => config.UseHttps(azureSettings.Certificate));
+            });
+
+            // Add services to the container.
+            builder.Services.AddControllers().AddJsonOptions(options =>
+            {
+                options.JsonSerializerOptions.WriteIndented = true; //pretty
+                options.JsonSerializerOptions.AllowTrailingCommas = true;
+                options.JsonSerializerOptions.PropertyNamingPolicy = JsonNamingPolicy.CamelCase;
+                options.JsonSerializerOptions.ReferenceHandler = ReferenceHandler.Preserve;
+                options.JsonSerializerOptions.Encoder = JavaScriptEncoder.UnsafeRelaxedJsonEscaping;
+                options.JsonSerializerOptions.Converters.Add(new ODataJsonConverterFactory(null, null, SerializerAssemblies.Assemblies));
+            });
 
-            var configuration = builder.Build();
+            var app = builder.Build();
 
-            ServiceCollection = new ServiceCollection();
-            ServiceCollection.AddCoreServices(configuration);
+            var host = new ServiceCollection().AddCoreServices(builder.Configuration);
 
-            ServiceProvider = ServiceCollection.BuildServiceProvider();
+            ServiceProvider = host.Build();
```

Comment:
how do the controllers access the services? where is the dependency injection connected to the Controllers?, do I miss something obvious here?

Reply:
Changed